src/Umbraco.Web/Security/IdentityFactoryMiddleware.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Owin;
+using Microsoft.Owin.Security.DataProtection;
+
+namespace Umbraco.Web.Security
+{
+    /// <summary>
+    /// Adapted from Microsoft.AspNet.Identity.Owin.IdentityFactoryMiddleware
+    /// </summary>
+    public class IdentityFactoryMiddleware<TResult, TOptions> : OwinMiddleware
+        where TResult : class, IDisposable
+        where TOptions : IdentityFactoryOptions<TResult>
+    {
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="next">The next middleware in the OWIN pipeline to invoke</param>
+        /// <param name="options">Configuration options for the middleware</param>
+        public IdentityFactoryMiddleware(OwinMiddleware next, TOptions options)
+            : base(next)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException("options");
+            }
+            if (options.Provider == null)
+            {
+                throw new ArgumentNullException("options.Provider");
```

Comment:
I guess this is more an `ArgumentException` then it is not the direct argument that is null.